rust/src/stracciatella.rs

```diff
@@ -193,7 +193,16 @@ fn parse_args(engine_options: &mut EngineOptions, args: Vec<String>) -> Option<S
 
             if let Some(s) = m.opt_str("datadir") {
                 match fs::canonicalize(PathBuf::from(s)) {
-                    Ok(s) => engine_options.vanilla_data_dir = s,
+                    Ok(s) => {
+                        let mut temp = String::from(s.to_str().expect("Should not happen"));
+                        // remove UNC path prefix (Windows)
+                        if temp.starts_with("\\\\") {
+                            temp.drain(..2);
+                            let pos = temp.find("\\").unwrap() + 1;
```

Comment:
Basically the unwrap denotes: I as a developer am absolutely sure that there wont be an erronous/non-existing value in there so you can just assume the value is ok.

If an error then occurs the whole application panics and exits. If you want to handle the non-existing case (in this case if the `\\` is not found, you can use either `match` to match for both cases or `if let` to handle a single case.